packages/context/src/reactServices/oidcUser.tsx

```diff
@@ -1,16 +1,23 @@
-import React, { ComponentType, PropsWithChildren, useContext } from 'react';
+import React, { useContext } from 'react';
 import { User } from 'oidc-client';
 
 import { AuthenticationContext } from '../oidcContext';
 
-type WithOidcUserComponentProps = PropsWithChildren<{ oidcUser: User }>;
+export type WithOidcUserProps = { oidcUser: User | null };
 
-export const withOidcUser = (Component: ComponentType<WithOidcUserComponentProps>) => (props: WithOidcUserComponentProps) => {
-  const { oidcUser } = useContext(AuthenticationContext);
-  const { children } = props;
-  return (
-    <Component {...props} oidcUser={oidcUser}>
-      {children}
-    </Component>
-  );
-};
+export function withOidcUser<T extends WithOidcUserProps = WithOidcUserProps>(
+  WrappedComponent: React.ComponentType<T>
+) {
+  // Try to create a nice displayName for React Dev Tools.
+  const displayName =
+    WrappedComponent.displayName || WrappedComponent.name || "Component";
+
+  const ComponentWithOidcUser = (props: Omit<T, keyof WithOidcUserProps>) => {
+    const { oidcUser } = useContext(AuthenticationContext);
+    return <WrappedComponent oidcUser={oidcUser} {...(props as T)} />;
+  };
+
+  ComponentWithOidcUser.displayName = `withOidcUser(${displayName})`;
+
+  return ComponentWithOidcUser;
```

Comment:
Are you sure this is working?

Reply:
Here is a screenshot of react-dev-tools for the context example. You can see the `withOidcUser` next to to `Admin`:
![exapple-context](https://user-images.githubusercontent.com/1098399/83965111-83ac3400-a8b1-11ea-8b8c-82236081df34.png)